youtube-viewer:  submission of new port 3.9.1

TheKevJames opened the pull request:
#### Description

Create new port for the `youtube-viewer` project. [Link](https://github.com/trizen/youtube-viewer).

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix
- [x] submission

###### Tested on

macOS 10.15.5 19F101 x86_64
Xcode 12.4 12D4e

###### Verification
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

reneeotten commented on multimedia/youtube-viewer/Portfile:
```diff
@@ -0,0 +1,74 @@
+    depends_lib-append \
+        port:p${perl5.major}-json-xs \
+        port:p${perl5.major}-lwp-useragent-cached
+}
```
please add the extra dependencies for the two variants inside of the variant declaration, that avoids needing to check if they're set here

reneeotten commented on multimedia/youtube-viewer/Portfile:
```diff
@@ -0,0 +1,74 @@
+perl5.create_variants   ${perl5.branches}
+
+name                youtube-viewer
+perl5.setup         youtube-viewer 3.9.1
```
please add a `revision 0` line under here

reneeotten reviewed (changes requested):
a few small comments. One thing to consider if variants are actually needed here (you anyway enable them by default). It seems that there are not many additional dependencies required to enable them and is there an obvious reason why an end-user would now want these features?

TheKevJames commented:
@reneeotten thank you for the review, I believe I've addressed all your comments. 

As for the variants, I'd be happy to remove them if you think it's reasonable -- since I am not the original author of youtube-viewer, I was mostly trying to stick with the original spirit of the original author's README, wherein these are considered [completely optional extras](https://github.com/trizen/youtube-viewer#optional-dependencies). I did consolidate them a bit to avoid too much complexity, with my goal being to strike a bit of a happy medium here. If it'd fit better for Macports to have everything installed by default, I can make that change, let me know!

I guess another thought I had here was to err away from homebrew's "we've picked the right set of dependencies" attitude here, which is what originally drove me to switch to macports. I don't know why an end-user might want a different set of variants -- but I'd assume that's their choice rather than being mine. Without a good reason otherwise, I would default to assuming that the variant system exists and we should thus make use of it: the default set of variants is what I consider reasonable but if, say, disabling local user agent caching is what a user wants then there's no need for me to remove that option from them.

reneeotten commented:
I don't know if there is an "official" MacPorts policy regarding when/how to use variants; it's probably up to the maintainers discretion to find a good balance. 

Personally, I would use them sparingly and only if they are for features that not many end-user will need by default and/or that require a large set of additional dependencies (and thus disk space) that, for example, due to licensing restrictions would not be distributable. In the latter case that would mean that there wouldn't be a MacPorts build binary and it would need to be built locally by the user. I think that's how variants are by and large used and things like `doc` and `native` or for different compilers / OpenMP / MPI.

Here we are talking about three [tiny (haven't checked)?] perl packages (or four if you include all of the optional ones); that include features that likely would improve the user experience and without an obvious drawback. So yes, I would just add all of them without the use of variants. If you prefer to keep the variants (but then perhaps find a better name than `perf`) that wouldn't stop this PR from being merged; anyway it can always changed later (although adding variants is always easier than removing them for compatibility reasons).

TheKevJames commented:
@reneeotten fair enough, that makes sense! I've gone ahead and flattened the variants into the main port, as requested.

reneeotten commented:
thanks @TheKevJames ! Last request (I promise 😉): are recent PR regarding Perl was merged removing 5.26 and adding 5.32; can you do that here as well? After doing so please squash your commits into one and I'll push the merge button.

TheKevJames commented:
No problem! Missed the perl update come through :) Should be good to go now :+1:

reneeotten commented:
thanks for the new port @TheKevJames and for your patience!
